start_admin_sshd.sh

```diff
@@ -108,11 +110,19 @@ for key_alg in rsa ecdsa ed25519; do
   if ssh-keygen -t "${key_alg}" -f "${SSH_HOST_KEY_DIR}/ssh_host_${key_alg}_key" -q -N ""; then
     chmod 600 "${SSH_HOST_KEY_DIR}/ssh_host_${key_alg}_key"
     chmod 644 "${SSH_HOST_KEY_DIR}/ssh_host_${key_alg}_key.pub"
+    ln -sf "${SSH_HOST_KEY_DIR}/ssh_host_${key_alg}_key.pub" "${SSHD_CONFIG_DIR}/ssh_host_${key_alg}_key.pub"
   else
     log "Failure to generate host ${key_alg} ssh keys"
   fi
 done
 
+if [[ ! -f "${SSH_HOST_KEY_DIR}/harvest" ]]; then
```

Comment:
In the case a user doesn't want to use EC2 Instance Connect, we're still going to run these commands and log a failure. I'm wondering if there is a better trigger or something we can key off of in order to run/not run these.

Reply:
Umm... are there any methods to fetch dynamic configuration from ec2 metadata or bottlerocket host such as `enable_eic: 1`?

Reply:
Currently we pass things to the admin container via the user-data json. At the moment it's only used for ssh keys, but perhaps we could also pass an enable-eic key/value pair. `imdsclient` and`shibaken` could potentially fetch it from IMDS. Anyone have any thoughts on that?

Reply:
```
declare -i eic_enabled=0
if ! eic_enabled=$(jq -e -r '.["enable_eic"]?' "${USER_DATA}"); then
  eic_enabled=1
fi
```
I just add this block to fetch "enable_eic" from userdata and make its default as 1 (if not exists); and execute the harvest script if and only if enable_eic=1.